assets/app/scripts/controllers/overview.js

```diff
@@ -58,6 +71,99 @@ angular.module('openshiftConsole')
       console.log("podsByServiceByLabel", $scope.podsByServiceByLabel);      
     };
 
+    var podRelationships = function() {
+      $scope.servicePodsNoDeployment = {};
+      $scope.solitaryPods = {};
+
+      angular.forEach($scope.pods, function(pod, name){
+        var replicationcontrollers = [];
+        var services = [];
+        angular.forEach($scope.deployments, function(deployment, depName){
+          var ls = new LabelSelector(deployment.spec.selector);
```

Comment:
add a TODO to construct the LabelSelectors first so we can reuse them rather than constructing them in an inner loop